srv/components/system/firewall/teardown.sls

```diff
@@ -53,7 +53,9 @@ Remove public and private data interfaces:
   cmd.run:
     - names:
       - firewall-cmd --remove-interface={{ pillar['cluster'][grains['id']]['network']['data_nw']['iface'][0] }} --zone=public-data-zone --permanent
+{%- if pillar['cluster'][node]['network']['data_nw']['iface']|length > 1 %}
```

Comment:
variable node is undefined in this file. Please use grains['id'] or define node variable first.